File: ExtremeParser/Parsers/Enums.vb
Imports System.Text.RegularExpressions

Public Class Enums
    Public Shared Sub Parse(ByRef code As String, fileName As String, ByRef parts As CodeParts, Optional add As Boolean = True)
        For Each Match As Match In Regex.Matches(code, "enum\s+([^\n;\(\)\{\}\s]*)\s+(?:(?:[{])([^}]+)(?:[}]))")
            Dim enumds As String() = Match.Groups(2).Value.Split(",")

            'Variable to store the enums contents.
            Dim enumStuff As New List(Of EnumsContentsClass)

            For Each enuma As String In enumds
                'Check if empty
                If enuma Is Nothing Or enuma.Trim = "" Then Continue For

                Dim length As Integer = enuma.Length + 1
                enuma = enuma.Trim
                Dim type = FunctionParameters.getVarType(enuma)

                'Do what needs to be changed
                If type = FunctionParameters.varTypes.TYPE_FLOAT Then
                    enuma = enuma.Remove(0, 6)
                ElseIf type = FunctionParameters.varTypes.TYPE_ARRAY Then
                    enuma = enuma.Remove(enuma.IndexOf("["), (enuma.IndexOf("]") - enuma.IndexOf("[")) + 1)
                ElseIf type = FunctionParameters.varTypes.TYPE_TAGGED Then
                    enuma = enuma.Remove(0, enuma.IndexOf(":") + 1)
                End If

                Try
                    enumStuff.Add(New EnumsContentsClass(enuma, type))
                Catch ex As Exception
                End Try
            Next

            'Now add it to the actual list.
            If add Then
                parts.Enums.Add(New EnumsStruct(Match.Groups(1).Value.Trim, enumStuff))
            Else
                parts.Enums.RemoveAll(Function(x) x.EnumName = Match.Groups(1).Value.Trim)
            End If
        Next
    End Sub
End Class

File: ExtremeParser/Parsers/Enums.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;
using ExtremeParser.Global;
using ExtremeParser.Types;

namespace ExtremeParser.Parsers
{
    public class Enums
    {
        public static void Parse(string code, string fileName, CodeParts parts, bool add = true)
        {
            foreach (Match match in Regex.Matches(code, "enum\\s+([^\\n;\\(\\)\\{\\}\\s]*)\\s+(?:(?:[{])([^}]+)(?:[}]))"))
            {
                string[] enumds = match.Groups[2].Value.Split(',');
			
                //Variable to store the enums contents.
                List<EnumsContentsClass> enumStuff = new List<EnumsContentsClass>();
			
                foreach (string enumatmp in enumds)
                {
                    string enuma = enumatmp;
                    //Check if empty
                    if (ReferenceEquals(enuma, null) || enuma.Trim() == "")
                    {
                        continue;
                    }
				
                    int length = enuma.Length + 1;
                    enuma = enuma.Trim();
                    var type = FunctionParameters.GetVarType(enuma);
				
                    //Do what needs to be changed
                    if (type == FunctionParameters.VarTypes.TypeFloat)
                    {
                        enuma = enuma.Remove(0, 6);
                    }
                    else if (type == FunctionParameters.VarTypes.TypeArray)
                    {
                        enuma = enuma.Remove(enuma.IndexOf("["), (enuma.IndexOf("]") - enuma.IndexOf("[")) + 1);
                    }
                    else if (type == FunctionParameters.VarTypes.TypeTagged)
                    {
                        enuma = enuma.Remove(0, enuma.IndexOf(":") + 1);
                    }
				
                    try
                    {
                        enumStuff.Add(new EnumsContentsClass(enuma, type));
                    }
                    catch (Exception)
                    {
                    }
                }
			
                //Now add it to the actual list.
                if (add)
                {
                    parts.Enums.Add(new EnumsStruct(match.Groups[1].Value.Trim(), enumStuff));
                }
                else
                {
                    parts.Enums.RemoveAll(x => x.EnumName == match.Groups[1].Value.Trim());
                }
            }
        }
    }
}
